frame/bridge/eth/offchain/src/lib.rs

```diff
@@ -203,21 +213,20 @@ impl<T: Trait> Module<T> {
 	/// if there are issue to communicate with scale encoding, the failback communication will
 	/// be performed with json format(use option: `true`)
 	fn relay_header() -> Result<(), DispatchError> {
-		if !T::SubmitSignedTransaction::can_sign() {
-			Err(<Error<T>>::AccountUnavail)?;
-		}
-
 		let target_number = Self::get_target_number()?;
 		let header_without_option = Self::fetch_header(ETHRESOURCE.to_vec(), target_number, false);
 		let (header, proof_list) = match header_without_option {
 			Ok(r) => r,
 			Err(e) => {
-				trace!(target: "eoc-rh", "[eth-offchain] request without option fail: {:?}", e);
-				trace!(target: "eoc-rh", "[eth-offchain] request fail back wth option");
+				info!(target: "eoc-rh", "[eth-offchain] request without option fail: {:?}", e);
+				info!(target: "eoc-rh", "[eth-offchain] request fail back wth option");
 				Self::fetch_header(ETHRESOURCE.to_vec(), target_number, true)?
 			}
 		};
 
+		// The `submit_header` will call event out of eth-offchain pallet,
+		// so this function is skiped in the test of pallet
+		#[cfg(not(test))]
```

Comment:
This looks tricky, should have better ways.

- we could put the tested code in the function to a seperate method.
- some other unit test might want to test the codes with ```submit_header```

Reply:
As discussed in zoom meeting, this is tricky but there is not realy big things need to be test in that function.  Please take a look and find out if it is acceptable.  Thanks.